### ISSUE TITLE
When GPU is enabled, are some new features of spark 3.0 still supported on rapids ？[QST]

### ISSUE THREAD

**Issue by rui911013** [NVIDIA/spark-rapids#384]
such as DPP（Dynamic partition pruning），AQE（Adaptive Query Execution），etc..

**Comment by revans2** [comment]
DPP will produce the correct answer, but we have not spent much time optimizing it.  The default DPP goes off of a broadcast join.  It will insert a sub-query expression into the filter expressions for the input format that is not being broadcast.  This all works as expected. Unfortunately we don't support sub-query expressions on the GPU yet, and even if we did, we would have to add in a special rule here. This means that the broadcast data for the sub-query is sent back to the CPU before it is broadcast.  Normally that is not a problem because that data is typically very small, but in this case it means that the broadcast exchange is not reused so we scan the broadcast table twice.  This is not ideal and I have filed an issue to look into how to optimize it.

https://github.com/NVIDIA/spark-rapids/issues/386


AQE is a harder problem with how the optimization engine interacts with the query.  @andygrove is working on it and is very close to having a solution, but to truly make it work we are going to need to push a few changes back into spark itself.  Hopefully we can get them in for the spark 3.1 release.
